### DISCUSSION TITLE
Does sqlalchemy supports chdb  serverless database

### DISCUSSION THREAD

**Discussion by flyly0755** [sqlalchemy/sqlalchemy#11319] [Usage Questions] [answered]
chdb comes from clickhouse, but only supports memory mode.
https://github.com/chdb-io/chdb

like sqlite memory mode, we can connect sqlite with sqlalchemy showed as below:
```python
from sqlalchemy import create_engine

# format 1, disk mode, uri = sqlite:///home/stephen/db1.db
engine = create_engine('sqlite:///home/stephen/db1.db')
# format 2,  memory mode, uri = sqlite:///:memory:
engine = create_engine('sqlite:///:memory:')

session = sessionmaker(bind=engine)()
```
So dose sqlalchemy supports chdb or not?
Seems need extra code development to support.
If need extra code development, which side burdens this major work? chdb or sqlalchemy?

**Comment by CaselIT** [comment, marked as answer]
Hi,

There is no dialect in sqlalchemy for it. There may be a community driver though, I have not looked.

> If need extra code development, which side burdens this major work? chdb or sqlalchemy?

I would say chdb. Sqlalchemy is not looking into supporting additional drivers directly
